### PR TITLE
Update sync.py

### PR DESCRIPTION
**Pull request by rderewianko**
Added an if statement for allowing invalid SSL. Forgive me this is me starting to dip my toes into python

### CODE REVIEW

File: sync.py [hunk 1]
```diff
@@ -157,6 +162,7 @@
     parser.add_argument('--password')
     parser.add_argument('--limit', type=int, default=25)
     parser.add_argument('--verbose', action='store_true')
+    parser.add_argument('--allowinvalidssl', action='store_true')
```
Comment (badstreff) [inline sync.py]: I would add a `default=False` param that way users have to explicitly disable it and it allows for some shorter code above - mainly we can just pass the param and not worry about an if-else block. It also allows us to be more explicit which is a good thing

File: sync.py [hunk 2]
```diff
@@ -143,9 +143,14 @@
 
 async def main(args):
     semaphore = asyncio.BoundedSemaphore(args.limit)
-    async with aiohttp.ClientSession() as session:
-        await upload_scripts(session, args.url, args.username, args.password, semaphore)
-        await upload_extension_attributes(session, args.url, args.username, args.password, semaphore)
+    if args.allowinvalidssl:
+        async with aiohttp.ClientSession(connector=aiohttp.TCPConnector(verify_ssl=False)) as session:
```
Comment (badstreff) [inline sync.py]: We can remove the if statement and just run these lines right here.

**Comment by badstreff** [comment]
Looking good! I did have some suggestions for cleaning it up, but I agree this would be a really helpful option!

File: sync.py (hunk 1 above)
Comment (rderewianko) [reply sync.py]: So this would become `parser.add_argument('--allowinvalidssl', action='store_true', default=False)`

File: sync.py (hunk 2 above)
Comment (rderewianko) [reply sync.py]: I think i'm more confused. Sorry
``` async def main(args):
    semaphore = asyncio.BoundedSemaphore(args.limit)
    async with aiohttp.ClientSession() as session:
        await upload_scripts(session, args.url, args.username, args.password, semaphore)
        await upload_extension_attributes(session, args.url, args.username, args.password, semaphore)
    if args.allowinvalidssl:
        async with aiohttp.ClientSession(connector=aiohttp.TCPConnector(verify_ssl=False)) as session:
            await upload_scripts(session, args.url, args.username, args.password, semaphore)
            await upload_extension_attributes(session, args.url, args.username, args.password, semaphore)
````
?

Comment (badstreff) [reply sync.py]: We should be able to:

```
async with aiohttp.ClientSession(connector=aiohttp.TCPConnector(verify_ssl=args.allowinvalidssl)) as session:
    await upload_scripts(session, args.url, args.username, args.password, semaphore)
    await upload_extension_attributes(session, args.url, args.username, args.password, semaphore)
```

And just have they switch with a default value

Comment (rderewianko) [reply sync.py]: Ahh... Okay, so I have a reason for preferring not a hard switch in the code like that.

In my environment, our DEV environment through a ci runner can have invalid ssl. Its a test environment torn away, and may have a self signed cert.

Production on the other hand, I want to verify that certificate. I want to only post to a server with a valid SSL config. Ergo, running the command as part of my runner, i can flag it off for dev. But not for prod.

Comment (badstreff) [reply sync.py]: ~~Right, my other comment where you added the param should solve that. Essential we want that param with a default value of True. Looks like I typed false by mistake~~

Whoops, meant to pass the args as the param to that function with a default value, we probably want to rename the arg though. verifyssl should work

I edited the code above to reflect what i was talking about

**Comment by rderewianko** [comment]
Thanks for your Patience! I used do_not_verify_ssl... more for the reasoning of I verifyssl defines to me more as "we're forcing on verification vs turning it off"

**Comment by badstreff** [comment]
Looks much better! Did you test in prod? We might still have to pass `default=True`

**Comment by rderewianko** [comment]
Sure did! Here's sample tests.
This test is without the flag.. It fails as expected.
```
python3 ./sync.py --url $casper_URL_DEV --username $casper_USER --password $casper_PASS --limit 15
Traceback (most recent call last):
  File "/home/gitlab-runner/.local/lib/python3.5/site-packages/aiohttp/connector.py", line 796, in _wrap_create_connection
    return (yield from self._loop.create_connection(*args, **kwargs))
  File "uvloop/loop.pyx", line 1769, in create_connection
  File "/usr/lib/python3.5/asyncio/futures.py", line 361, in __iter__
    yield self  # This tells Task to wait for completion.
  File "/usr/lib/python3.5/asyncio/tasks.py", line 296, in _wakeup
    future.result()
  File "/usr/lib/python3.5/asyncio/futures.py", line 274, in result
    raise self._exception
  File "/usr/lib/python3.5/asyncio/sslproto.py", line 493, in data_received
    ssldata, appdata = self._sslpipe.feed_ssldata(data)
  File "/usr/lib/python3.5/asyncio/sslproto.py", line 201, in feed_ssldata
    self._sslobj.do_handshake()
  File "/usr/lib/python3.5/ssl.py", line 633, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:645)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "./sync.py", line 164, in <module>
    loop.run_until_complete(main(args))
  File "uvloop/loop.pyx", line 1364, in uvloop.loop.Loop.run_until_complete
  File "/usr/lib/python3.5/asyncio/futures.py", line 274, in result
    raise self._exception
  File "/usr/lib/python3.5/asyncio/tasks.py", line 241, in _step
    result = coro.throw(exc)
  File "./sync.py", line 148, in main
    await upload_scripts(session, args.url, args.username, args.password, semaphore)
  File "./sync.py", line 92, in upload_scripts
    responses = await asyncio.gather(*tasks)
  File "/usr/lib/python3.5/asyncio/futures.py", line 361, in __iter__
    yield self  # This tells Task to wait for completion.
  File "/usr/lib/python3.5/asyncio/tasks.py", line 296, in _wakeup
    future.result()
  File "/usr/lib/python3.5/asyncio/futures.py", line 274, in result
    raise self._exception
  File "/usr/lib/python3.5/asyncio/tasks.py", line 241, in _step
    result = coro.throw(exc)
  File "./sync.py", line 105, in upload_script
    auth=auth) as resp:
  File "/home/gitlab-runner/.local/lib/python3.5/site-packages/aiohttp/client.py", line 690, in __aenter__
    self._resp = yield from self._coro
  File "/home/gitlab-runner/.local/lib/python3.5/site-packages/aiohttp/client.py", line 267, in _request
    conn = yield from self._connector.connect(req)
  File "/home/gitlab-runner/.local/lib/python3.5/site-packages/aiohttp/connector.py", line 402, in connect
    proto = yield from self._create_connection(req)
  File "/home/gitlab-runner/.local/lib/python3.5/site-packages/aiohttp/connector.py", line 748, in _create_connection
    _, proto = yield from self._create_direct_connection(req)
  File "/home/gitlab-runner/.local/lib/python3.5/site-packages/aiohttp/connector.py", line 859, in _create_direct_connection
    raise last_exc
  File "/home/gitlab-runner/.local/lib/python3.5/site-packages/aiohttp/connector.py", line 831, in _create_direct_connection
    req=req, client_error=client_error)
  File "/home/gitlab-runner/.local/lib/python3.5/site-packages/aiohttp/connector.py", line 801, in _wrap_create_connection
    raise ClientConnectorSSLError(req.connection_key, exc) from exc
aiohttp.client_exceptions.ClientConnectorSSLError: Cannot connect to host <devserverhere>:8443 ssl:True [[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:645)]
ERROR: Job failed: exit status 1
```
This test is run with the flag, as expected passes.

```
python3 ./sync.py --url $casper_URL_DEV --username $casper_USER --password $casper_PASS --limit 15 --do_not_verify_ssl
Uploaded script: Machine Scripts - adbind.sh 
<more scripts here...>
Job succeeded
```

Reading up on `action='store_false'` defines it as true until invoked.

**Pull request merged**